### DISCUSSION TITLE
Can't rewrite root path ('/') to function when output contains `index.html`.

### DISCUSSION THREAD

**Discussion by damienwebdev** [vercel/vercel#5723] [Help] [answered]
```
{
  "version": 2,
  "public": true,
  "functions": {
    "api/serverless.js": {
      "runtime": "@vercel/node@1.8.5"
    }
  },
  "rewrites": [
    { "source": "", "destination": "api/serverless.js"}, //doesn't work
    { "source": "/(.*)", "destination": "api/serverless.js" },
    { "source": "/(.*[.].+)", "destination": "/$1"}
  ]
}
```

I have a vercel.json like the above, but it appears that my root is instead being treated as the static file `index.html` that's included in my output folder. Is there a way to fix this?

**Comment by damienwebdev** [comment, marked as answer]
This is partially answered by https://github.com/vercel/vercel/discussions/5303 , but given that `routes` is now legacy, how should this be handled?

**Reply by damienwebdev** [reply]
@styfle

**Reply by styfle** [reply]
This is expected since static files take precedence over `rewrites`.

Consider renaming your `index.html` file if you don't want it served on the home page.

**Reply by Patrick-ring-motive** [reply]
It is really annoying that this issue exists. I am converting my apps from using the build directive in vercel.json to using the function directive so that I can tune the memory allocation. These are plain nodejs apps with no framework. What happens is everything works swimmingly except at the very top level '/' where my 'index.mjs' is served up in plain text. It isn't a security issue for me because my code is public but I can see this being an issue for many. Here is the workaround that I came up with:

 I set the 'content-type' header to 'text/html' within vercel.json specifically for '/'

I added the page that I actually want served under '/_root' from nodejs. It is dynamic so I can't just hard code the page. Then added this to the top of index.mjs

```

/** 
Root Fallback
<style>
*{display:none;}
</style>
<script>
void async function(){

let root=await fetch('/_root');
let homePage=await root.text();

document.write(homePage);

}();
</script>
*/

```
 
This works fine but it is pretty silly that I have to do this.

**Comment by 269652** [comment]
Has this been solved? I want to deploy a SSR Vite application and the express route for '/' is not being called because Vercel serves the index.html in the root folder of the source. 

How can I work around this? 

1. a way to exclude the index.html (note that the build step requires the file) so .vercelignore doesn't help
2. somehow don't output the source files at all. output only the files generated during the build step
3. Put the entire project in a sub folder and have the build step generate files outside of it so that no static files reside in the root directory which is an absolutely hacky workaround
4. disable static file serving in vercel.json
